packages/core/src/root.ts

```diff
@@ -215,6 +229,40 @@ export function createRemoteRoot<
 
       return text;
     },
+    createFragment() {
+      const id = `${currentId++}`;
+
+      const internals: FragmentInternals = {
+        children: strict ? Object.freeze([]) : [],
+      };
+
+      const fragment: RemoteFragment<Root> = {
+        kind: KIND_FRAGMENT,
+        get children() {
+          return internals.children;
+        },
+        appendChild: (child) =>
+          appendChild(
+            fragment,
+            normalizeChild(child, remoteRoot),
+            internals,
+            rootInternals,
+          ),
+        removeChild: (child) =>
+          removeChild(fragment, child, internals, rootInternals),
+        insertChildBefore: (child, before) =>
+          insertChildBefore(fragment, child, before, internals, rootInternals),
+
+        // Just satisfying the type definition, since we need to write
+        // some properties manually.
+        ...EMPTY_OBJECT,
+      };
+
+      makePartOfTree(fragment, rootInternals);
```

Comment:
Even it's not `corrected` to the tree however it still needs to be attached to something so that we can keep track of it and send back to the receiver. A fragment prop is currently attached to the component holding it (similar to a child but not actually a child). A sub tree is attached to the fragment.

Reply:
Resolve this conversation because of this https://github.com/Shopify/remote-ui/pull/73#discussion_r625980506